extensions/arc/src/models/postgresModel.ts

```diff
@@ -103,4 +126,84 @@ export class PostgresModel extends ResourceModel {
 			this._refreshPromise = undefined;
 		}
 	}
+
+	public async getEngineSettings(): Promise<void> {
+		if (!this._connectionProfile) {
+			await this.getConnectionProfile();
+		}
+
+		// We haven't connected yet so do so now and then store the ID for the active connection
+		if (!this._activeConnectionId) {
+			const result = await azdata.connection.connect(this._connectionProfile!, false, false);
+			if (!result.connected) {
+				throw new Error(result.errorMessage);
+			}
+			this._activeConnectionId = result.connectionId;
+		}
+
+		const provider = azdata.dataprotocol.getProvider<azdata.QueryProvider>(this._connectionProfile!.providerName, azdata.DataProviderType.QueryProvider);
+		const ownerUri = await azdata.connection.getUriForConnection(this._activeConnectionId);
+
+		const engineSettings = await provider.runQueryAndReturn(ownerUri, 'select name, setting, short_desc,min_val, max_val, enumvals, vartype from pg_settings');
+		if (!engineSettings) {
+			throw new Error('Could not fetch engine settings');
+		}
+		engineSettings.rows.forEach(row => {
+			let rowValues = row.map(c => c.displayValue);
+			let result: EngineSettingsModel = {
+				parameterName: rowValues.shift(),
+				value: rowValues.shift(),
+				description: rowValues.shift(),
+				min: rowValues.shift(),
+				max: rowValues.shift(),
+				options: rowValues.shift(),
+				type: rowValues.shift()
+			};
+
+			this._engineSettings.push(result);
```

Comment:
Does this duplicate the _engineSettings array each time settings are refreshed?  Does it need to clear the existing array before pushing the latest values?  Or do something like this?:

`this._engineSettings = engineSettings.rows.map(row=>...)`

Reply:
Yeah that was caught and corrected in next PR. I clear the array before it pushes back into it.